ObjectiveCloudantTests/QueryViewTests.m

```diff
@@ -0,0 +1,971 @@
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+}
+
+- (void)testSuccessfullyQueryViewNoOptions
+{
+    XCTestExpectation *requestCompleted =
+        [self expectationWithDescription:@"Succesfully view query"];
+    XCTestExpectation *viewRowBlockCalled = [self expectationWithDescription:@"Document Found"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+
+    __block int numberOfDocs = 0;
+    __block BOOL firstDoc = YES;
+    op.viewRowBlock = ^(NSDictionary<NSString *, NSObject *> *doc) {
+      if (firstDoc) {
+          firstDoc = NO;
+          [viewRowBlockCalled fulfill];
+      }
+      numberOfDocs++;
+      XCTAssertNotNil(doc);
+      XCTAssertNotNil(doc[@"id"]);
+      XCTAssertNotNil(doc[@"key"]);
+      XCTAssertNotNil(doc[@"value"]);
+      XCTAssertNil(doc[@"doc"]);
+    };
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [requestCompleted fulfill];
+      XCTAssertNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+
+    XCTAssertEqual(10, numberOfDocs);
+}
+
+- (void)testSuccessfullyQueryViewLimit1
+{
+    XCTestExpectation *requestCompleted =
+        [self expectationWithDescription:@"Succesfully view query"];
+    XCTestExpectation *viewRowBlockCalled = [self expectationWithDescription:@"Document Found"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.limit = 1;
+
+    op.viewRowBlock = ^(NSDictionary<NSString *, NSObject *> *doc) {
+      [viewRowBlockCalled fulfill];
+      XCTAssertNotNil(doc);
+      XCTAssertNotNil(doc[@"id"]);
+      XCTAssertNotNil(doc[@"key"]);
+      XCTAssertNotNil(doc[@"value"]);
+      XCTAssertEqualObjects(@"kookaburra", doc[@"id"]);
+      XCTAssertEqualObjects(@"carnivore", doc[@"key"]);
+      XCTAssertEqualObjects(@1, doc[@"value"]);
+    };
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [requestCompleted fulfill];
+      XCTAssertNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+}
+
+- (void)testSuccessfullyQueryViewDescending
+{
+    NSArray *expectedOrderOfKeys = @[
+        @"omnivore",
+        @"omnivore",
+        @"omnivore",
+        @"omnivore",
+        @"herbivore",
+        @"herbivore",
+        @"herbivore",
+        @"herbivore",
+        @"carnivore",
+        @"carnivore",
+    ];
```

Comment:
Do you need to test that they are ascending if you don't apply the descending argument (i.e., could this set of tests pass even if `op.descending` is always set to `YES`?